ramsete-controller/robot.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) FIRST and other WPILib contributors.
+# Open Source Software; you can modify and/or share it under the terms of
+# the WPILib BSD license file in the root directory of this project.
+#
+
+import wpilib
+import wpimath.controller
+import wpimath.filter
+import wpimath.geometry
+import wpimath.trajectory
+import wpimath.units
+
+from drivetrain import Drivetrain
+
+
+class MyRobot(wpilib.TimedRobot):
+    """
+    Uses the CameraServer class to automatically capture video from a USB webcam and send it to the
+    FRC dashboard without doing any vision processing. This is the easiest way to get camera images
+    to the dashboard. Just add this to the robotInit() method in your program.
+    """
+
+    def robotInit(self):
+        self.controller = wpilib.XboxController(0)
+        self.drive = Drivetrain()
+
+        # Slew rate limiters to make joystick inputs more gentle; 1/3 sec from 0 to 1.
+        self.speedLimiter = wpimath.filter.SlewRateLimiter(3)
+        self.rotLimiter = wpimath.filter.SlewRateLimiter(3)
+
+        # An example trajectory to follow during the autonomous period.
+        self.trajectory = wpimath.trajectory.TrajectoryGenerator.generateTrajectory(
+            wpimath.geometry.Pose2d(0, 0, wpimath.geometry.Rotation2d.fromDegrees(0)),
+            [
+                wpimath.geometry.Translation2d(1, 1),
+                wpimath.geometry.Translation2d(2, -1),
+            ],
+            wpimath.geometry.Pose2d(3, 0, wpimath.geometry.Rotation2d.fromDegrees(0)),
+            wpimath.trajectory.TrajectoryConfig(
+                wpimath.units.feetToMeters(3.0), wpimath.units.feetToMeters(3.0)
+            ),
+        )
+
+        # The Ramsete Controller to follow the trajectory.
+        self.ramseteController = wpimath.controller.RamseteController()
+
+        # The timer to use during the autonomous period.
+        self.timer = wpilib.Timer()
+
+        # Create Field2d for robot and trajectory visualizations.
+        self.field = wpilib.Field2d()
+
+        # Create and push Field2d to SmartDashboard.
+        wpilib.SmartDashboard.putData(self.field)
+
+        # Push the trajectory to Field2d.
+        self.field.getObject("traj").setTrajectory(self.trajectory)
+
+    def autonomousInit(self):
+        # Initialize the timer.
+        self.timer = wpilib.Timer()
```

Comment:
Don't need to recreate the timer here.

Reply:
I believe the java example uses that as a way to re-initialize the timer.  Should I replace it with `timer.reset()` or just remove the line entirely?